sycl/doc/SYCL_compiler_and_runtime_design.md

```diff
@@ -0,0 +1,377 @@
+
+will produce binaries from SYCL kernels for devices identified by the two
+target triples. This basically tells the driver which device compilers must be
+invoked to compile the SYCL kernel code. By default, the JIT compilation
+approach is assumed and device code is compiled for a single target triple -
+`[spir,spir64]-\*-\*`.
+
+#### Device code formats
+
+Each target may support a number of code forms, each device compiler defines
+and understands mnemonics designating a particular code form, for example
+"`visa:3.3`" could designate virtual ISA version 3.3 for Intel GPU target (Gen
+architecture).  User can specify desired code format using the target-specific
+option mechanism, similar to OpenMP.
+
+`-Xsycl-target=<triple> <arg>`
+
+For example, to support offload to CPU, FPGA, Gen9/vISA3.3, Gen9/SPIR-V the
+following options would be used:
+
+`-fsycl -fsycl-targets=x86,fpga,gen9 -Xsycl-target=gen9 "-fmt:visa -fmt:spirv"`
+
+The `<arg>` parameter is passed by the driver directly to the SYCL device
+compiler for the corresponding target w/o parsing it. For each target there is
+some default code form which is generated in the absence of overriding via the
+`-Xsycl-target` option.
+
+**TBD:** Having multiple code forms for the same target in the fat binary might
+mean invoking device compiler multiple times. Multiple invocations are not
+needed if these forms can be dumped at various compilation stages by the single
+device compilation, like SPIR-V->visa->ISA. But if e.g. `gen9:visa3.2` and
+`gen9:visa3.3` are needed at the same time, then some mechanism is needed.
+Should it be a dedicated target triple for each needed visa version or Gen
+generation?
+
+#### Separate Compilation and Linking
+
+The compiler supports linking of device code obtained from different source
+files before generating the final SPIR-V to be fed to the back-end. The basic
+mechanism is to produce "fat objects" as a result of compilation - object files
+containing both host and device code for all targets - then break fat objects
+into their constituents before linking and link host code and device code
+(per-target) separately and finally produce a "fat binary" - a host executable
+with embedded linked images for each target specified at the command line.
+
+![Multi source compilation flow](Multi-source-compilation-flow.png)
+
+The clang driver orchestrates compilation and linking process based on a
+SYCL-specific offload action builder and invokes external tools as needed. On
+the diagram above, every dark-blue box is a tool invoked as a separate process
+by the clang driver.
+
+Compilation starts with compiling the input source `a.cpp` for one of the
+targets requested via the command line - `T2`. When doing this first
+compilation, the driver requests the device compiler to generate an
+"integration header" via a special option. Device compilation for other targets
+\- `T1` - don't need to generate the integration header, as it must be the same
```

Comment:
I guess that if you replace these `-` by some `—` then you do not need to escape them anymore.